extension/packages/nextjs/app/eip-712/page.tsx

```diff
@@ -0,0 +1,154 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import Link from "next/link";
+import type { NextPage } from "next";
+import useSWRMutation from "swr/mutation";
+import { SignTypedDataReturnType } from "viem/accounts";
+import { useAccount, useSignTypedData, useVerifyTypedData } from "wagmi";
+import {
+  EIP_712_DOMAIN,
+  EIP_712_TYPE,
+  VerifyRequestBody,
+  generateMessageToBob,
+} from "~~/utils/eip-712";
+import { notification } from "~~/utils/scaffold-eth";
+import { postMutationFetcher } from "~~/utils/swr";
+
+const Home: NextPage = () => {
+  const { address: connectedAddress } = useAccount();
+  const [signature, setSignature] = useState<SignTypedDataReturnType>();
+  const { signTypedDataAsync } = useSignTypedData();
+
+  const [name, setName] = useState("");
+  const [message, setMessage] = useState("");
+
+  const { trigger: verify } = useSWRMutation(
+    "/api/verify",
+    postMutationFetcher<VerifyRequestBody>
+  );
+
+  const typedData = {
+    domain: EIP_712_DOMAIN,
+    types: EIP_712_TYPE,
+    primaryType: "Mail",
+    message: generateMessageToBob({
+      fromName: name,
+      fromAddress: connectedAddress,
+      message,
+    }),
+  } as const;
+
+  const { data: verifiedOnFrontend } = useVerifyTypedData({
+    ...typedData,
+    address: connectedAddress,
+    signature,
+  });
+
+  const signTypedData = async () => {
+    const signature = await signTypedDataAsync(typedData);
+
+    setSignature(signature);
+    return signature;
+  };
```

Comment:
Maybe wrap inside `try / catch` because we user reject it on Metamask we get unhandled error : 
<details>

<summary>
Example : 
</summary>

![Screenshot 2024-06-13 at 12 10 05 PM](https://github.com/scaffold-eth/create-eth-extensions/assets/80153681/15c315eb-4a50-494f-8d7e-8b186e32bcac)


</details>

```ts
  const signTypedData = async () => {
    try {
      const signature = await signTypedDataAsync(typedData);
      setSignature(signature);
      return signature;
    } catch (e) {
      const errorMessage = getParsedError(e);
      notification.error(errorMessage);
    }
  };

```

Also I propose using `getParsedError` utility function while handling, since it will nicely parse error of different kind and fallbacks to `e.message`  so it will work always

Reply:
Thanks, updated [981741a](https://github.com/scaffold-eth/create-eth-extensions/pull/3/commits/981741a88e885d1ec1be0d103f325a5bbbbe1211)